internal/collector/otel_collector_plugin.go

```diff
@@ -62,26 +66,24 @@ func (oc *Collector) Init(ctx context.Context, mp bus.MessagePipeInterface) erro
 		return fmt.Errorf("write OTel Collector config: %w", err)
 	}
 
-	go func() {
-		bootErr := oc.bootup(runCtx)
-		if bootErr != nil {
-			slog.ErrorContext(runCtx, "error", err)
-		}
-	}()
+	bootErr := oc.bootup(runCtx)
+	if bootErr != nil {
+		slog.ErrorContext(runCtx, "Unable to start OTel Collector", "error", bootErr)
+	}
 
 	return nil
 }
 
 func (oc *Collector) bootup(ctx context.Context) error {
-	oc.appDone = make(chan struct{})
 	errChan := make(chan error)
 
+	oc.stopped = false
```

Comment:
Just a side note, should we toggle this to `true` when we receive an error from `oc.service.Run(ctx)` and/or when state is `otelcol.StateClosed` in the for loop below?